src/channel.ts

```diff
@@ -1351,6 +1362,10 @@ export class Channel<StreamChatGenerics extends ExtendableGenerics = DefaultGene
         break;
       case 'channel.updated':
         if (event.channel) {
+          const isFrozenChanged = event.channel?.frozen !== undefined && event.channel.frozen !== channel.data?.frozen;
+          if (isFrozenChanged) {
```

Comment:
```suggestion
          const isFrozen = event.channel?.frozen !== undefined && event.channel.frozen !== channel.data?.frozen;
          if (isFrozen) {
```

Since we only query when channel is frozen, not the other way round.

Reply:
We need to query if frozen changes from `false` -> `true`, but also if it changed from `true` -> `false`. So we want to detect state changes for the `frozen` flag, so I think the name is correct here.

Reply:
Should messages limit be 0? It will just wipe the messages from UI. Is that expected?

Reply:
It doesn't wipe messages, at least not from `channel.state`, it only adds new messages from the query result: https://github.com/GetStream/stream-chat-js/blob/master/src/channel.ts#L1474

Reply:
messages will be an empty array in the `channels.queried` event's payload: https://github.com/GetStream/stream-chat-js/blob/master/src/channel.ts#L1035 - but not in `channel.state`

Reply:
The reason `limit: 0` is set is to avoid unnecessarily loading the same messages that are already in the state, I've only tested with Angular SDK, but I didn't spot any issue there, let me know if you see it otherwise.

Reply:
I went though the `_initializeState` and this is the only wiping out possibility I've spotted:https://github.com/GetStream/stream-chat-js/pull/1196/commits/63fdfabcd1f071bd17e619bf381e3abf2961d9f6